05-Introducing-Rigid-Body-Molecular-Dynamics/02-Initializing-a-Random-System.ipynb

```diff
@@ -0,0 +1,390 @@
+    "\n",
+    "We assume the radius of the particle is the half the minimum distance between two LJ spheres at minimum pairwise energy. \n",
+    "\n",
+    "$r_m = 2^{1/6}\\sigma$\n",
+    "\n",
+    "We therefore set $r_m$ to 2 and therefore $\\sigma$ equals $2/2^{1/6}$\n",
+    "\n",
+    "We also assume $\\epsilon = 1.$\n",
+    "\n",
+    "# Parameters of the Integrator:\n",
+    "\n",
+    "An important thing to note is that we must tell hoomd to integrate the rotational degrees of freedom, otherwise our rigid body will not change from the initial orientation. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "f0974f8e-62d8-44b8-a356-5a8df1db1443",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "epsilon = 1\n",
+    "sigma = 2/2**(1/6)\n",
+    "integrator = hoomd.md.Integrator(dt=0.0001,integrate_rotational_dof=True)\n",
+    "cell = hoomd.md.nlist.Cell(buffer=0,exclusions = ['body'])\n",
+    "lj = hoomd.md.pair.LJ(nlist=cell)\n",
+    "lj.params[('rigid_body', 'rigid_body')] = dict(epsilon=0, sigma=sigma)\n",
+    "lj.params[('rigid_body', 'constituent_sphere')] = dict(epsilon=0, sigma=sigma)\n",
+    "lj.params[('constituent_sphere', 'constituent_sphere')] = dict(epsilon=epsilon, sigma=sigma)\n",
+    "lj.r_cut[('constituent_sphere', 'constituent_sphere')] = 4 * sigma \n",
+    "lj.r_cut[('rigid_body', 'rigid_body')] = 0.\n",
+    "lj.r_cut[('rigid_body', 'constituent_sphere')] = 0.\n",
+    "integrator.forces.append(lj)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "92568aa8-b342-4044-a3c5-282995191066",
+   "metadata": {},
+   "source": [
+    "## Adding rigid body constraint:\n",
+    "\n",
+    "This adds the rigid bodies to the simulations and marks them as 'body' type particles. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "4fd8f444-47ef-4678-b483-4efe99243c0d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rigid = constrain.Rigid()\n",
+    "\n",
+    "constituent_types = ['constituent_sphere','constituent_sphere']\n",
+    "\n",
+    "rigid.body['rigid_body'] = {\n",
+    "    \"constituent_types\": constituent_types,\n",
+    "    \"positions\":local_position,\n",
+    "    \"orientations\":[(1,0,0,0)] * len(constituent_types),\n",
+    "    \"charges\":[0.] * len(constituent_types),\n",
+    "    \"diameters\":[2.] * len(constituent_types)}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c4bab758-8031-41a2-b8f3-0684cab1291c",
+   "metadata": {},
+   "source": [
+    "# Filtering integration in hoomd:\n",
+    "\n",
+    "Hoomd must filter out the integration of only the rigid bodies and has a built in method to do so.\n",
+    "\n",
+    "We must apply this method in our Nose-Hoover NVT integrator. \n",
+    "\n",
+    "*Center* particles are denoted as the rigid body centers defined above. *free* particles are particles that are neither a rigid body center, nor constituent particles of a rigid body.  "
```

Comment:
First time you use rigid body center. It would be helpful to mention it earlier.